tests/yam/agama/agama.pm

```diff
@@ -0,0 +1,49 @@
+## Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Run interactive installation with Agama,
+# run playwright tests directly from the Live ISO.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base Yam::agama::agama_base;
+use strict;
+use warnings;
+
+use testapi qw(
+  assert_screen
+  assert_script_run
+  select_console
+  get_required_var
+  upload_logs
+  match_has_tag
+  reset_consoles
+  get_var
+);
+
+sub run {
+    my $test = get_required_var('AGAMA_TEST');
+    my $cmd;
+    if ($test !~ /take_screenshots/) {
+        $cmd = 'PRODUCTNAME="' . get_required_var('AGAMA_PRODUCT') . '"';
+    } else {
+        $cmd = 'RUN_INSTALLATION=1';
+    }
+    assert_script_run($cmd . ' playwright test --trace on --project chromium --config /usr/share/e2e-agama-playwright tests/' . $test . '.spec.ts', timeout => 1200);
+
+    $testapi::password = 'nots3cr3t';
+    assert_script_run('reboot');
+    # For agama test, it is too short time to match the grub2, so we create
+    # a new needle to avoid too much needles loaded.
+    if (!get_var('ENCRYPT')) {
+       assert_screen('grub2-agama', 120);
+       my @tags = ("welcome-to", "login");
+       assert_screen(\@tags, 300);
+       reset_consoles;
+    }
+}
+
+sub post_fail_hook {
+    upload_logs("./test-results/agama/trace.zip");
```

Comment:
I thing the name of the directory will be agama-name of the test from inside the playwright code-chromium. I assume we could change the playwright settings to save logs in a specific folder

Reply:
Here we can simply to ls 'test-results' to get the trace log name, then upload the file is ok. One thing need to consider is there maybe not only one file. Besides, this upload trace file needed not only when failure happened but also passed for sometimes we'd like to get detail info during the test via view the trace log.

Reply:
I was checking and didn't find a way to set a specific path, looks like it is like that by design, in case of multiple traces.

Reply:
I guess we would need two commands, one to archive all the trace.zip there and then upload that archive, so when we archive we can pass a regex and avoid to write specific things related with test name or its description, wdyt?

Reply:
Yes, something like 
find ./test-results '*trace.zip*' -exec mv -t log_path {} +

Reply:
@sofiasyria could you please file a ticket to achieve this, it is going to be too much here.